test/test_c10d.py

```diff
@@ -72,6 +72,22 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def retry_on_address_already_in_use_error(func):
+    """Reruns a test if it sees "Address already in use" error."""
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        for _ in range(10):
+            try:
+                return func(*args, **kwargs)
```

Comment:
add a small timeout of 100ms or so, otherwise it's probably going to be useless.

Reply:
I think it shouldn't matter TBH, because it will try to find a new port every time. But it doesn't hurt and this path should almost never hit anyway, so I'll add a random sleep to add some jitter as well.